### PR TITLE
Check Referer and Origin headers for Cross-Site Request Forgery protection

### PR DESCRIPTION
**Pull request by jsharkey13**

### CODE REVIEW

File: src/main/java/uk/ac/cam/cl/dtg/segue/api/managers/UserAuthenticationManager.java [hunk 1]
```diff
@@ -319,6 +323,23 @@ public RegisteredUser getUserFromSession(final HttpServletRequest request) {
             return null;
         }
 
+        if (properties.getProperty(Constants.SEGUE_APP_ENVIRONMENT).equals(EnvironmentType.PROD.name())) {
```
Comment (jsharkey13) [inline src/main/java/uk/ac/cam/cl/dtg/segue/api/managers/UserAuthenticationManager.java]: I wonder whether checking this once on Manager creation and storing the boolean as a class attribute would be better than checking every request?

Comment (sacummins) [reply src/main/java/uk/ac/cam/cl/dtg/segue/api/managers/UserAuthenticationManager.java]: Possibly, but I think the properties all get preloaded into a map in memory anyway so it is probably not much difference.

File: src/main/java/uk/ac/cam/cl/dtg/segue/api/managers/UserAuthenticationManager.java [hunk 2]
```diff
@@ -319,6 +323,23 @@ public RegisteredUser getUserFromSession(final HttpServletRequest request) {
...
+            // If in production, check if the request originated from Isaac:
+            String referrer = request.getHeader("Referer");  // Note HTTP Header misspelling!
+            if (null == referrer) {
+                log.warn("Authenticated request had no 'Referer' information set!");
```
Comment (sacummins) [inline src/main/java/uk/ac/cam/cl/dtg/segue/api/managers/UserAuthenticationManager.java]: Given that you are logging these to try and help find out in what circumstances they happen, would it be useful to log the request url seeing as you have the request object in scope?

Comment (jsharkey13) [reply src/main/java/uk/ac/cam/cl/dtg/segue/api/managers/UserAuthenticationManager.java]: I think this is a good idea!

File: src/main/java/uk/ac/cam/cl/dtg/segue/api/managers/UserAuthenticationManager.java (hunk 1 above)
Comment (jsharkey13) [reply src/main/java/uk/ac/cam/cl/dtg/segue/api/managers/UserAuthenticationManager.java]: In which case I will leave it as-is!

File: src/main/java/uk/ac/cam/cl/dtg/segue/api/managers/UserAuthenticationManager.java (hunk 2 above)
Comment (sacummins) [reply src/main/java/uk/ac/cam/cl/dtg/segue/api/managers/UserAuthenticationManager.java]: Did you want the attempted to access part on this line too or just the ones below?

Comment (jsharkey13) [reply src/main/java/uk/ac/cam/cl/dtg/segue/api/managers/UserAuthenticationManager.java]: It never happend :P
The magic of rebasing! Good spot again though, thanks for checking carefully!

**Pull request merged**